boto3/dynamodb/table.py

```diff
@@ -144,14 +144,12 @@ def _flush(self):
         response = self._client.batch_write_item(
             RequestItems={self._table_name: items_to_send}
         )
-        unprocessed_items = response['UnprocessedItems']
-
-        if unprocessed_items and unprocessed_items[self._table_name]:
-            # Any unprocessed_items are immediately added to the
-            # next batch we send.
-            self._items_buffer.extend(unprocessed_items[self._table_name])
-        else:
-            self._items_buffer = []
+        unprocessed_items = response['UnprocessedItems'].get(
+            self._table_name, []
```

Comment:
@nateprewitt do you know of an event where response would be falsey? At least according to previously existing tests if there are no `UnprocessedItems` that key would still exist. If response can be `{}` or `None` I'll add an `if` back in.

Reply:
It _shouldn't_ be `None` but we were previously accounting for that. I think the simplest approach is to do a quick check and convert if we encounter it.

i.e.

```suggestion
        if not unprocessed_items:
            unprocessed_items = {}
        unprocessed_items = response['UnprocessedItems'].get(
            self._table_name, []
```

Reply:
done. thanks Nate!